software/common/libs/framing/frame_detector.tpp

```diff
@@ -0,0 +1,165 @@
+/* Copyright 2020-2022, RespiraWorks
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+////////////////////////////////////////////////////////////////////////////////
+/// WARNING: Do not include this file as header, nor compile it as a source! ///
+///                                                                          ///
+/// This is a template implementation file.                                  ///
+/// It is included by the corresponding header.                              ///
+////////////////////////////////////////////////////////////////////////////////
+
+template<size_t S>
```

Comment:
I am curious of why templates are used here and not a class with a suitable 
constructor. The parameter <S> seems not to be used, or am I wrong?